--- a/chrome/browser/ui/views/tabs/common/tab_view.cc	2026-09-11 16:08:21
+++ b/chrome/browser/ui/views/tabs/common/tab_view.cc	2026-09-11 16:08:10
@@ -528,12 +528,23 @@
 }
 
 void TabView::OnPaint(gfx::Canvas* canvas) {
-  // Split pinned tabs have a merged background that is rendered in
-  // `SplitTabView`.
   if (pinned_ && split_) {
     return;
   }
 
+  if (GetColorProvider()) {
+    cc::PaintFlags flags;
+    flags.setAntiAlias(true);
+    if (active_ || selected_) {
+      flags.setColor(GetColorProvider()->GetColor(ui::kColorSysPrimary));
+    } else {
+      flags.setColor(GetColorProvider()->GetColor(ui::kColorSysSurfaceVariant));
+    }
+    canvas->DrawRoundRect(gfx::RectF(GetLocalBounds()), pinned_ ? 0.0f : 8.0f, flags);
+    views::View::OnPaint(canvas);
+    return;
+  }
+
   if (active_tab_fill_id_.has_value() || inactive_tab_fill_id_.has_value()) {
     PaintTabBackgroundWithImages(canvas, active_tab_fill_id_,
                                  inactive_tab_fill_id_);
