--- a/mojo/core/ipcz_driver/wrapped_platform_handle.cc
+++ b/mojo/core/ipcz_driver/wrapped_platform_handle.cc
@@ -24,8 +24,12 @@
 
 #if BUILDFLAG(IS_APPLE)
 #include <mach/mach.h>
-#include <sys/fileport.h>
 
+extern "C" {
+int fileport_makefd(mach_port_t);
+kern_return_t fileport_makeport(int, mach_port_t*);
+}
+
 #include "base/apple/mach_logging.h"
 #include "base/apple/scoped_mach_port.h"
 #endif

--- a/mojo/core/channel_mac.cc
+++ b/mojo/core/channel_mac.cc
@@ -6,8 +6,12 @@
 
 #include <mach/mach.h>
 #include <string.h>
-#include <sys/fileport.h>
 #include <unistd.h>
+
+extern "C" {
+int fileport_makefd(mach_port_t);
+kern_return_t fileport_makeport(int, mach_port_t*);
+}
 
 #include <algorithm>
 #include <memory>
