--- a/third_party/blink/renderer/platform/mac/graphics_context_canvas.mm
+++ b/third_party/blink/renderer/platform/mac/graphics_context_canvas.mm
@@ -8,6 +8,10 @@
 #import <AppKit/AppKit.h>
 #import <CoreGraphics/CoreGraphics.h>
 
+#ifndef kCGImageByteOrder32Host
+#define kCGImageByteOrder32Host kCGBitmapByteOrder32Host
+#endif
+
 #include "base/apple/scoped_cftyperef.h"
 #include "cc/paint/paint_canvas.h"
 #include "skia/ext/skia_utils_mac.h"
