--- a/device/fido/mac/icloud_keychain.mm
+++ b/device/fido/mac/icloud_keychain.mm
@@ -6,6 +6,27 @@
 
 #import <AuthenticationServices/AuthenticationServices.h>
 #import <Foundation/Foundation.h>
+
+#if !defined(MAC_OS_VERSION_15_0) || MAC_OS_X_VERSION_MAX_ALLOWED < 150000
+@interface ASAuthorizationPublicKeyCredentialPRFRegistrationOutput : NSObject
+@property(readonly, nonatomic) BOOL isSupported;
+@property(readonly, nonatomic, nullable) NSData* first;
+@property(readonly, nonatomic, nullable) NSData* second;
+@end
+
+@interface ASAuthorizationPublicKeyCredentialPRFAssertionOutput : NSObject
+@property(readonly, nonatomic, nullable) NSData* first;
+@property(readonly, nonatomic, nullable) NSData* second;
+@end
+
+@interface ASAuthorizationPlatformPublicKeyCredentialRegistration (MacOS15Compatibility)
+@property(readonly, nonatomic, nullable) ASAuthorizationPublicKeyCredentialPRFRegistrationOutput* prf;
+@end
+
+@interface ASAuthorizationPlatformPublicKeyCredentialAssertion (MacOS15Compatibility)
+@property(readonly, nonatomic, nullable) ASAuthorizationPublicKeyCredentialPRFAssertionOutput* prf;
+@end
+#endif
 
 #include <algorithm>
 #include <optional>

--- a/device/fido/mac/icloud_keychain_sys.mm
+++ b/device/fido/mac/icloud_keychain_sys.mm
@@ -5,6 +5,31 @@
 #include "device/fido/mac/icloud_keychain_sys.h"
 
 #import <AuthenticationServices/AuthenticationServices.h>
+
+#if !defined(MAC_OS_VERSION_15_0) || MAC_OS_X_VERSION_MAX_ALLOWED < 150000
+@interface ASAuthorizationPublicKeyCredentialPRFAssertionInputValues : NSObject
+- (instancetype)initWithSaltInput1:(NSData*)saltInput1
+                        saltInput2:(nullable NSData*)saltInput2;
+@end
+
+@interface ASAuthorizationPublicKeyCredentialPRFRegistrationInput : NSObject
++ (instancetype)checkForSupport;
+- (instancetype)initWithInputValues:(ASAuthorizationPublicKeyCredentialPRFAssertionInputValues*)inputValues;
+@end
+
+@interface ASAuthorizationPublicKeyCredentialPRFAssertionInput : NSObject
+- (instancetype)initWithInputValues:(nullable ASAuthorizationPublicKeyCredentialPRFAssertionInputValues*)inputValues
+           perCredentialInputValues:(nullable NSDictionary<NSData*, ASAuthorizationPublicKeyCredentialPRFAssertionInputValues*>*)perCredentialInputValues;
+@end
+
+@interface ASAuthorizationPlatformPublicKeyCredentialRegistrationRequest (MacOS15Compatibility)
+@property(nonatomic, strong, nullable) ASAuthorizationPublicKeyCredentialPRFRegistrationInput* prf;
+@end
+
+@interface ASAuthorizationPlatformPublicKeyCredentialAssertionRequest (MacOS15Compatibility)
+@property(nonatomic, strong, nullable) ASAuthorizationPublicKeyCredentialPRFAssertionInput* prf;
+@end
+#endif
 
 #include "base/functional/callback.h"
 #include "base/memory/scoped_refptr.h"
