--- a/third_party/crashpad/crashpad/util/mach/exc_server_variants.cc
+++ b/third_party/crashpad/crashpad/util/mach/exc_server_variants.cc
@@ -114,19 +114,23 @@
   using ExceptionRaiseStateRequest = __Request__mach_exception_raise_state_t;
   using ExceptionRaiseStateIdentityRequest =
       __Request__mach_exception_raise_state_identity_t;
+#if defined(EXCEPTION_STATE_IDENTITY_PROTECTED)
   using ExceptionRaiseIdentityProtectedRequest =
       __Request__mach_exception_raise_identity_protected_t;
   using ExceptionRaiseStateIdentityProtectedRequest =
       __Request__mach_exception_raise_state_identity_protected_t;
+#endif
 
   using ExceptionRaiseReply = __Reply__mach_exception_raise_t;
   using ExceptionRaiseStateReply = __Reply__mach_exception_raise_state_t;
   using ExceptionRaiseStateIdentityReply =
       __Reply__mach_exception_raise_state_identity_t;
+#if defined(EXCEPTION_STATE_IDENTITY_PROTECTED)
   using ExceptionRaiseIdentityProtectedReply =
       __Reply__mach_exception_raise_identity_protected_t;
   using ExceptionRaiseStateIdentityProtectedReply =
       __Reply__mach_exception_raise_state_identity_protected_t;
+#endif
 
   // The MIG-generated __MIG_check__Request__*() functions are not declared as
   // accepting const data, but they could have been because they in fact do not
@@ -154,6 +158,7 @@
         const_cast<ExceptionRaiseStateIdentityRequest**>(in_request_1));
   }
 
+#if defined(EXCEPTION_STATE_IDENTITY_PROTECTED)
   static kern_return_t MIGCheckRequestExceptionRaiseIdentityProtected(
       const ExceptionRaiseIdentityProtectedRequest* in_request) {
     return __MIG_check__Request__mach_exception_raise_identity_protected_t(
@@ -170,23 +175,28 @@
             const_cast<ExceptionRaiseStateIdentityProtectedRequest**>(
                 in_request_1));
   }
+#endif
 
   // There are no predefined constants for these.
   static const mach_msg_id_t kMachMessageIDExceptionRaise = 2405;
   static const mach_msg_id_t kMachMessageIDExceptionRaiseState = 2406;
   static const mach_msg_id_t kMachMessageIDExceptionRaiseStateIdentity = 2407;
+#if defined(EXCEPTION_STATE_IDENTITY_PROTECTED)
   static const mach_msg_id_t kMachMessageIDExceptionRaiseIdentityProtected =
       2408;
   static const mach_msg_id_t
       kMachMessageIDExceptionRaiseStateIdentityProtected = 2410;
+#endif
 
   static std::set<mach_msg_id_t> RequestIDs() {
     return {
         kMachMessageIDExceptionRaise,
         kMachMessageIDExceptionRaiseState,
         kMachMessageIDExceptionRaiseStateIdentity,
+#if defined(EXCEPTION_STATE_IDENTITY_PROTECTED)
         kMachMessageIDExceptionRaiseIdentityProtected,
         kMachMessageIDExceptionRaiseStateIdentityProtected,
+#endif
     };
   }
 
@@ -274,6 +284,7 @@
         const mach_msg_trailer_t* trailer,
         bool* destroy_request) = 0;
 
+#if defined(EXCEPTION_STATE_IDENTITY_PROTECTED)
     //! \brief Handles exceptions raised by
     //!     `mach_exception_raise_identity_protected()`.
     //!
@@ -320,6 +331,7 @@
         mach_msg_type_number_t* new_state_count,
         const mach_msg_trailer_t* trailer,
         bool* destroy_request) = 0;
+#endif
 
    protected:
     ~Interface() {}
@@ -486,6 +498,7 @@
   }
 
   if constexpr (std::is_same_v<Traits, MachExcTraits>) {
+#if defined(EXCEPTION_STATE_IDENTITY_PROTECTED)
     switch (in_header->msgh_id) {
       case Traits::kMachMessageIDExceptionRaiseIdentityProtected: {
         // mach_exception_raise_identity_protected(),
@@ -689,6 +702,7 @@
         return true;
       }
     }
+#endif
   }
 
   SetMIGReplyError(out_header, MIG_BAD_ID);
@@ -846,6 +860,7 @@
         destroy_request);
   }
 
+#if defined(EXCEPTION_STATE_IDENTITY_PROTECTED)
   kern_return_t CatchExceptionRaiseIdentityProtected(
       exception_handler_t exception_port,
       thread_t thread,
@@ -904,6 +919,7 @@
         trailer,
         destroy_request);
   }
+#endif
 
  private:
   Interface* interface_;  // weak
