--- a/ui/events/keycodes/keyboard_code_conversion_mac.mm
+++ b/ui/events/keycodes/keyboard_code_conversion_mac.mm
@@ -143,7 +143,7 @@
       {kVK_RightArrow, DomKey::ARROW_RIGHT},
       {kVK_DownArrow, DomKey::ARROW_DOWN},
       {kVK_UpArrow, DomKey::ARROW_UP},
-      {kVK_ContextualMenu, DomKey::CONTEXT_MENU},
+      {0x6E, DomKey::CONTEXT_MENU},
       {kVK_JIS_Eisu, DomKey::EISU},
       {kVK_JIS_Kana, DomKey::KANJI_MODE},
   });
