--- a/chrome/browser/ui/views/tabs/tab.cc
+++ b/chrome/browser/ui/views/tabs/tab.cc
@@ -1,4 +1,15 @@
 
 void Tab::OnPaint(gfx::Canvas* canvas) {
+  if (GetColorProvider()) {
+    cc::PaintFlags flags;
+    flags.setAntiAlias(true);
+    if (IsActive() || IsSelected()) {
+      flags.setColor(GetColorProvider()->GetColor(ui::kColorSysPrimary));
+    } else {
+      flags.setColor(GetColorProvider()->GetColor(ui::kColorSysSurfaceVariant));
+    }
+    canvas->DrawRoundRect(gfx::RectF(GetLocalBounds()), 6.0f, flags);
+    return;
+  }
   tab_style_views()->PaintTab(canvas);
 }
